firedrake/slate/slate.py

```diff
@@ -664,21 +670,16 @@ def _split_arguments(self):
         """Splits the function space and stores the component
         spaces determined by the indices.
         """
-        from firedrake.functionspace import FunctionSpace, MixedFunctionSpace
-        from firedrake.ufl_expr import Argument
-
         tensor, = self.operands
         nargs = []
         for i, arg in enumerate(tensor.arguments()):
             V = arg.function_space()
-            V_is = V.subfunctions
-            idx = as_tuple(self._blocks[i])
+            idx = self._blocks[i]
             if len(idx) == 1:
-                fidx, = idx
-                W = V_is[fidx]
+                W = V[idx[0]]
                 W = FunctionSpace(W.mesh(), W.ufl_element())
             else:
-                W = MixedFunctionSpace([V_is[fidx] for fidx in idx])
+                W = MixedFunctionSpace([V[fidx] for fidx in idx])
 
             nargs.append(Argument(W, arg.number(), part=arg.part()))
```

Comment:
Using `subspace` this whole function could just be be replaced with
```suggestion
         return (Argument(subspace(V, self._blocks[i]), arg.number(), part=arg.part())
                 for i, arg in enumerate(self.operands[0].arguments()))
```